.buildkite/commands/configure-git-for-release-management.sh

```diff
@@ -3,11 +3,6 @@
 # The Git command line client is not configured in Buildkite.
 # At the moment, steps that need Git access can configure it on deman using this script.
 # Later on, we should be able to configure it on the agent instead.
-
-curl -L https://api.github.com/meta | jq -r '.ssh_keys | .[]' | sed -e 's/^/github.com /' >> ~/.ssh/known_hosts
+add_host_to_ssh_known_hosts github.com
 git config --global user.email "mobile+wpmobilebot@automattic.com"
 git config --global user.name "Automattic Release Bot"
-
-# Buildkite is currently using the HTTPS URL to checkout.
-# We need to override it to be able to use the deploy key.
-git remote set-url origin git@github.com:Automattic/simplenote-ios.git
```

Comment:
The previous version is how WordPress does it, https://github.com/wordpress-mobile/WordPress-iOS/blob/b4de5f928cd784cb85144ca469b27910666d8d30/.buildkite/commands/configure-git-for-release-management.sh, this how WooCommerce does it, https://github.com/woocommerce/woocommerce-ios/blob/c2f38422ddb12b891afbab9962869affa7722af2/.buildkite/commands/configure-git-for-release-management.sh

Given Woo is a later iteration, I decided to adopt it. 

_I also secretly hope it will help with the Git access that prompted this PR when we'll get to run the complete code freeze, but I don't see how._

Reply:
> _I also secretly hope it will help with the Git access that prompted this PR when we'll get to run the complete code freeze, but I don't see how._

As expected, this wasn't enough. 

But, one only had to read the error in CI more closely to understand what the problem was:

```
ERROR: The key you are authenticating with has been marked as read only.
```

Unsurprisingly, the deploy key in this repo was read-only!

Once replaced with a read-write one, everything worked. :okay:

Reply:
@mokagio  But wasn't the whole point to keep the deploy key set in GitHub as read-only for security purposes? And use `use-bot-for-git.sh` on a trusted agent for the push action?

Reply:
True; the deploy key of the repo should be used only for cloning the repo in non-release pipelines (therefore being read-only). In release pipelines (which at some point should run on a trusted agent) such as the one calling `start_code_freeze`, we wouldn't use the repo deploy key at all.

Reply:
@iangmaia @AliSoftware `start_code_freeze` runs on the trusted agent:

https://github.com/Automattic/simplenote-ios/blob/92fd936a500ed11ec70dbfc16c13c4318350d065/.buildkite/release-pipelines/start-code-freeze.yml#L12-L13

The problem is with [`complete_code_freeze`](https://github.com/Automattic/simplenote-ios/blob/92fd936a500ed11ec70dbfc16c13c4318350d065/.buildkite/release-pipelines/complete-code-freeze.yml), which needs to run on macOS because it accesses `genstrings`.